src/main/java/com/aizistral/nochatreports/common/mixins/client/MixinChatComponent.java

```diff
@@ -62,6 +68,9 @@ private FormattedText modifyGUIMessage(FormattedText msg) {
 					Component.Serializer.toJson((Component) msg));
 		}
 
+		if(msg.isSigned) //TODO: no idea lol
```

Comment:
TODO: should first mixin GuiMessage, which has a parameter messageSignature. That can be validated with MessageSignature.verify() or SignatureValidator.validate().